src/vhost_user/slave_req_handler.rs

```diff
@@ -399,12 +399,12 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
                 self.send_reply_message(&hdr, &msg)?;
             }
             MasterReq::SET_VRING_ENABLE => {
-                let msg = self.extract_request_body::<VhostUserVringState>(&hdr, size, &buf)?;
-                if self.acked_protocol_features & VhostUserProtocolFeatures::MQ.bits() == 0
-                    && msg.index > 0
+                if self.acked_virtio_features & VhostUserVirtioFeatures::PROTOCOL_FEATURES.bits()
+                    == 0
                 {
                     return Err(Error::InvalidOperation);
                 }
+                let msg = self.extract_request_body::<VhostUserVringState>(&hdr, size, &buf)?;
```

Comment:
Should we receive the VhostUserVringState message before checking the PROTOCOL_FEATURES flag? Otherwise there will be a malformed message body left in the communication pipe.

Reply:
Good catch! Thanks.
Fixed.